### ISSUE TITLE
lost_pn can go negative

### ISSUE THREAD

**Issue by larseggert** [quicwg/base-drafts#2611]
in `DetectLostPackets`, `lost_pn` is computed as
```
lost_pn = largest_acked_packet[pn_space] - kPacketThreshold
```
However, `largest_acked_packet` is initialized to zero
```
largest_acked_packet[pn_space] = 0
```
and so the subtraction can cause `lost_pn` to go negative when `DetectLostPackets` is called before an ACK is received (since it will be updated only in `OnAckReceived`).

Suggest to initialize `largest_acked_packet` to infinity and to explicitly check for this case.

**Comment by marten-seemann** [comment]
I support this proposal, since the way we’re initializing largest_acked_packet is wrong: we’re initializing it to 0, although we decided that 0 is a valid packet number.

**Comment by ianswett** [comment]
Closed by #2621

**Comment by ianswett** [comment]
I merged #2611 into a random local branch instead of head, so I'm reopening this.
